include/xtensor/xrandom.hpp

```diff
@@ -0,0 +1,146 @@
+        {
+            get_default_random_engine().seed(seed);
+        }
+    }
+
+    /**
+     * @function rand
+     * @brief xexpression with specified @p shape containing uniformly distributed random numbers 
+     *        in the interval from @p lower to @p upper, excluding upper.
+     * 
+     * Numbers are drawn from @c std::uniform_real_distribution.
+     * 
+     * @param shape shape of resulting xexpression
+     * @param lower lower bound
+     * @param upper upper bound
+     * @param engine random number engine
+     *
+     * @tparam T number type to use
+     */ 
+    template <class T, class E = random::default_engine_type>
+    inline auto rand(std::vector<std::size_t> shape, T lower = 0, T upper = 1,
+                     E& engine = random::get_default_random_engine())
+    {
+        std::uniform_real_distribution<T> dist(lower, upper);
+        return detail::make_xgenerator(detail::random_impl<T>(std::bind(dist, std::ref(engine))), shape);
+    }
+    
+    /**
+     * @function randint
+     * @brief xexpression with specified @p shape containing uniformly distributed 
+     *        random integers in the interval from @p lower to @p upper, excluding upper.
+     * 
+     * Numbers are drawn from @c std::uniform_int_distribution.
+     * 
+     * @param shape shape of resulting xexpression
+     * @param lower lower bound
+     * @param upper upper bound
+     * @param engine random number engine
+     *
+     * @tparam T number type to use
+     */
+    template <class T, class E = random::default_engine_type>
+    inline auto randint(std::vector<std::size_t> shape,
+                        T lower = 0, T upper = std::numeric_limits<T>::max(), 
+                        E& engine = random::get_default_random_engine())
+    {
+        std::uniform_int_distribution<T> dist(lower, upper - 1);
+        return detail::make_xgenerator(detail::random_impl<T>(std::bind(dist, std::ref(engine))), shape);
+    }
+
+    /**
+     * @function randn
+     * @brief xexpression with specified @p shape containing numbers sampled from 
+     *        the Normal (Gaussian) random number distribution with mean @p mean and 
+     *        standard deviation @p std_dev.
+     * 
+     * Numbers are drawn from @c std::normal_distribution.
+     * 
+     * @param shape shape of resulting xexpression
+     * @param mean mean of normal distribution
+     * @param std_dev standard deviation of normal distribution
+     * @param engine random number engine
+     * 
+     * @tparam T number type to use
+     */
+    template <class T, class E = random::default_engine_type>
+    inline auto randn(std::vector<std::size_t> shape,
+                      T mean = 0, T std_dev = 1,
+                      E& engine = random::get_default_random_engine())
+    {
+        std::normal_distribution<T> dist(mean, std_dev);
+        return detail::make_xgenerator(detail::random_impl<T>(std::bind(dist, std::ref(engine))), shape);
+    }
+}
```

Comment:
there was a brief discussion with @JohanMabille about the new namespace. I think he was against introducing more namespaces. Personally, I'd like to follow numpy here, and add a random namespace.